go/obscuronode/enclave/db/storage.go

```diff
@@ -199,16 +199,25 @@ func (s *storageImpl) FetchBlockState(hash obscurocommon.L1RootHash) (*core.Bloc
 	return nil, false
 }
 
-func (s *storageImpl) SetBlockState(hash obscurocommon.L1RootHash, state *core.BlockState, rollup *core.Rollup) {
-	if state.Block != hash {
-		log.Panic("failed sanity check: `state.Block.Hash() != hash`")
-	}
+func (s *storageImpl) SaveNewHead(state *core.BlockState, rollup *core.Rollup, receipts []*types.Receipt) {
+	batch := s.db.NewBatch()
 
 	if state.FoundNewRollup {
-		s.StoreRollup(rollup)
+		obscurorawdb.WriteRollup(batch, rollup)
+		obscurorawdb.WriteCanonicalHash(batch, rollup.Hash(), rollup.NumberU64())
+		obscurorawdb.WriteHeadHeaderHash(batch, rollup.Hash())
+		obscurorawdb.WriteCanonicalHash(batch, rollup.Hash(), rollup.NumberU64())
```

Comment:
@tudor-malene Is this duplication of writing the canonical hash intentional?

Reply:
no. good spot.